### PR TITLE
[WIP] Conversion to revscriptsys and client id

### PR DESCRIPTION
**Pull request by dudantas**
* Revscriptsys conversion:
- [x] Convert actions to revscriptsys and client id (tested)
- [x] Convert creaturescript to revscriptsys and client id (tested
- [x] Convert globalevents to revscriptsys and client id (tested)
- [x] Convert items.xml from server id to client id (tested)
- [x] Convert items.otb to client id (tested)
- [x] Convert map to client id (tested, but I will still make some corrections)
- [ ] Convert monster to client id and full revision (in progress)
- [x] Convert movements to revscriptsys and client id (tested)
- [x] Convert NPC's to client id (tested)
- [x] Convert talkactions to revscriptsys (tested)
- [ ] Convert spells to revscriptsys and client id
- [x] Convert weapons to revscriptsys and client id (tested)

* Features/Systems
- [x] Review doors script
- [x] Review windows script
- [ ] Remove old XML load (if required/want)
- [ ] Remove action/unique from the map and move to Lua (should be discussed, if this change is accepted, I can do the commit)
Example: https://github.com/opentibiabr/otservbr-global/blob/develop/data/lib/tables/action_unique.lua
I understand that this is a very complex change, if accepted, it must be loaded through the source, right after the map is loaded, and not by globalevents as it is in the OTBR repository.

This pull request probably has a high chance of not being accepted, however, hese are significant advances and have cost us many months...

[Download Tool](https://github.com/opentibiabr/tools/blob/master/Convert%20MAP%20to%20Client%20ID.rar) for convert map to client id.

Everything is being tested (of course it is not free of errors)
I will give the necessary support to the PR, as long as it is within my reach. I am totally open to suggestions.

### CODE REVIEW

**Comment by mattyx14** [comment]
for to otx example convert owns map to clientid how?

**Comment by dudantas** [comment]
> for to otx example convert owns map to clientid how?

Yes ... There are a few ways to convert.

Particularly I think that a function should be added in the rme of saving in client Id. Or created some tool to convert it into client Id.

File: data/global.lua [hunk 1]
```diff
@@ -1,15 +1,7 @@
...
-questDoors = {1223, 1225, 1241, 1243, 1255, 1257, 3542, 3551, 5105, 5114, 5123, 5132, 5288, 5290, 5745, 5748, 6202, 6204, 6259, 6261, 6898, 6907, 7040, 7049, 8551, 8553, 9175, 9177, 9277, 9279, 10278, 10280, 10475, 10484, 10782, 10791, 12097, 12104, 12193, 12202, 19847, 19856, 19987, 19996, 20280, 20289, 25162, 25164}
-levelDoors = {1227, 1229, 1245, 1247, 1259, 1261, 3540, 3549, 5103, 5112, 5121, 5130, 5292, 5294, 6206, 6208, 6263, 6265, 6896, 6905, 7038, 7047, 8555, 8557, 9179, 9181, 9281, 9283, 10282, 10284, 10473, 10482, 10780, 10789, 10780, 12095, 12102, 12204, 12195, 19845, 19854, 19985, 19994, 20278, 20287}
-keys = {2086, 2087, 2088, 2089, 2090, 2091, 2092, 10032}
+ropeSpots = {386, 421, 12935, 12936, 14238, 17238, 21501, 21965, 21966, 21967, 21968, 23363}
```
Comment (EPuncker) [inline data/global.lua:4]: new rope spots removed...

File: data/global.lua [hunk 2]
```diff
@@ -68,3 +60,7 @@ end
 if not nextUseStaminaTime then
 	nextUseStaminaTime = {}
 end
+
+function isNumber(str)
```
Comment (EPuncker) [inline data/global.lua:64]: this looks like belonging to compat.lua

Comment (dudantas) [reply data/global.lua]: Moved

File: data/global.lua (hunk 1 above)
Comment (dudantas) [reply data/global.lua]: 8278 is duplicated cid 386
image

The rest I added now
Thanks

**Comment by dudantas** [comment]
> for to otx example convert owns map to clientid how?

I added in the pull request description a tool to convert the map into a client ID

**Pull request closed without merging**